api/integrations/github/github.py

```diff
@@ -78,25 +86,38 @@ def generate_body_comment(
     if is_removed:
         return delete_text
 
-    last_updated_string = LAST_UPDATED_FEATURE_TEXT % (
-        datetime.datetime.now().strftime("%dth %b %Y %I:%M%p")
+    last_updated_string = datetime.datetime.now().strftime(
+        get_format("DATETIME_INPUT_FORMATS")[0]
     )
-    updated_text = UPDATED_FEATURE_TEXT % (name)
 
-    result = updated_text if is_update else LINK_FEATURE_TEXT % (name)
+    result = UPDATED_FEATURE_TEXT % (name) if is_update else LINK_FEATURE_TITLE % (name)
+    last_segment_name = ""
+    if len(feature_states) > 0 and not feature_states[0].get("segment_name"):
+        result += FEATURE_TABLE_HEADER
 
-    # if feature_states is None:
     for v in feature_states:
         feature_value = v.get("feature_state_value")
-        feature_value_type = v.get("feature_state_value_type")
-
-        feature_value_string = f"\n{feature_value_type}\n```{feature_value if feature_value else 'No value.'}```\n\n"
-
-        result += f"**{v['environment_name']}{' - ' + v['segment_name'] if v.get('segment_name') else ''}**\n"
-        result += f"- [{'x' if v['feature_value'] else ' '}] {'Enabled' if v['feature_value'] else 'Disabled'}"
-        result += feature_value_string
+        tab = "segment-overrides" if v.get("segment_name") is not None else "value"
+        environment_link_url = FEATURE_ENVIRONMENT_URL % (
+            get_current_site_url(),
+            project_id,
+            v.get("environment_api_key"),
+            feature_id,
+            tab,
+        )
+        if v.get("segment_name") is not None and v["segment_name"] != last_segment_name:
+            result += "\n" + LINK_SEGMENT_TITLE % (v["segment_name"])
+            last_segment_name = v["segment_name"]
+            result += FEATURE_TABLE_HEADER
```

Comment:
We should add a segment to the test or slap a `# pragma: no cover` on L108.

Reply:
Done.